db/migrate/20170508085402_add_not_null_constraint_to_sources_payment_type.rb

```diff
@@ -0,0 +1,11 @@
+class AddNotNullConstraintToSourcesPaymentType < ActiveRecord::Migration
+  def change
+    reversible do |dir|
+      dir.up do
+        SolidusPaypalBraintree::Source.where(payment_type: nil).
+          update_all(payment_type: 'CreditCard')
```

Comment:
Yes. All credit card payments. PayPal and ApplePay payments have this value set via JS tailored params. 

Although this isn't accurate it's the only thing we can do about, besides not doing this at all of course.